src/RestApi/StoreApi/Routes/CartRemoveCoupon.php

```diff
@@ -69,9 +60,14 @@ protected function get_route_post_response( \WP_REST_Request $request ) {
 		$controller  = new CartController();
 		$cart        = $controller->get_cart_instance();
 		$coupon_code = wc_format_coupon_code( $request['code'] );
+		$coupon      = new \WC_Coupon( $coupon_code );
+
+		if ( $coupon->get_code() !== $coupon_code || ! $coupon->is_valid() ) {
+			throw new RouteException( 'woocommerce_rest_cart_coupon_error', __( 'Invalid coupon code.', 'woo-gutenberg-products-block' ), 403 );
+		}
 
 		if ( ! $controller->has_coupon( $coupon_code ) ) {
-			throw new RouteException( 'woocommerce_rest_cart_coupon_invalid_code', __( 'Coupon does not exist in the cart.', 'woo-gutenberg-products-block' ), 404 );
+			throw new RouteException( 'woocommerce_rest_cart_coupon_invalid_code', __( 'Coupon is not applied to the cart.', 'woo-gutenberg-products-block' ), 409 );
```

Comment:
The wording is better than what it was but it still feels kinda clunky. Maybe `Coupon did not apply to the cart.`?

Reply:
I will reword for clarity but this is the remove endpoint, so it's not that it did not apply, but that it cannot be removed because it was not already applied.